File: problems/11. Container With Most Water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height)-1
        result = 0
        while left < right:
            a = min(height[left], height[right]) * (right - left)
            result = max(result, a)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return result

File: problems/11. Container With Most Water/solution.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int result = 0;
        int left = 0;
        int right = height.size() - 1;
        while (left < right)
        {
            int area = std::min(height[left], height[right]) * (right - left);
            result = std::max(result, area);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }

        return result;
    }
};
